app/src/main/java/com/kms/cura/view/fragment/PatientHomeFragment.java

```diff
@@ -0,0 +1,211 @@
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
+import android.widget.Spinner;
+
+import com.kms.cura.R;
+import com.kms.cura.controller.ErrorController;
+import com.kms.cura.model.Settings;
+import com.kms.cura.utils.GPSTracker;
+import com.kms.cura.view.UpdateSpinner;
+import com.kms.cura.view.adapter.CheckBoxAdapter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class PatientHomeFragment extends Fragment implements RadioGroup.OnCheckedChangeListener ,UpdateSpinner{
+
+    private EditText edtName, edtLocation;
+    private RadioGroup rdbtngroupLocation;
+    private RadioButton rdbtnCurrentLocation, rdbtnManualEnter;
+    private Spinner spnSpeciality;
+    private Button btnRegister;
+    private Context mContext;
+    private Activity activity;
+    private String currentLocation = null;
+    private boolean checked = true;
+    private ArrayList<String> speciality;
+    private UpdateSpinner updateSpinner;
+    private CheckBoxAdapter specialityAdapter;
+    private boolean[] specialitySelected = null; // it stores which specialities are chosen
+
+
+    public PatientHomeFragment() {
+        // Required empty public constructor
+    }
+
+    public static PatientHomeFragment newInstance(Context mContext, Activity activity) {
+        PatientHomeFragment fragment = new PatientHomeFragment();
+        fragment.setContext(mContext);
+        fragment.setActivity(activity);
+        return fragment;
+    }
+
+    private void setContext(Context src) {
+        this.mContext = src;
+    }
+
+    private void setActivity(Activity src) {
+        this.activity = src;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View root = inflater.inflate(R.layout.fragment_patient_home, container, false);
+        updateSpinner = this;
+        initView(root);
+        setUpSpnSpeciality(root);
+        return root;
+    }
+
+    public void initView(View root) {
+        edtName = (EditText) root.findViewById(R.id.edtName);
+        edtLocation = (EditText) root.findViewById(R.id.edtLocation);
+        rdbtngroupLocation = (RadioGroup) root.findViewById(R.id.rdbtngroupLoacation);
+        rdbtngroupLocation.setOnCheckedChangeListener(PatientHomeFragment.this);
+        rdbtnCurrentLocation = initRadioButton(root, R.id.rdbtnCurrentLocation, 1);
+        rdbtnManualEnter = initRadioButton(root, R.id.rdbtnManuallyEnter, 2);
+    }
+
+    public RadioButton initRadioButton(View root, int id, int newId) {
+        RadioButton tmp = (RadioButton) root.findViewById(id);
```

Comment:
should not name variable as tmp, instead name it by their type, like in this case you should name it "radioButton"